import-automation/import-progress-dashboard-frontend/client/src/import-table.tsx

```diff
@@ -0,0 +1,378 @@
+    // Array of log IDs. The actual logs can be retrieved from 
+    // /import_attempts/{attempt_id}/logs
+    logs: Array<string>
+}
+
+/**
+ * A log can be linked to a system run, an import attempt, or both.
+ */
+type ProgressLog = {
+    log_id: string,
+    time_logged: string,
+    level: string,
+    message: string
+}
+
+
+type ImportAttemptRowProps = {
+    attempt: ImportAttempt
+}
+
+type ImportAttemptRowState = {
+    // Whether the row is collapsed
+    open: boolean,
+    logs: Array<ProgressLog>
+}
+
+/**
+ * A row for an import attempt. The row can be collapsed to show the logs of the
+ * import attempt.
+ */
+class ImportAttemptRow extends React.Component<ImportAttemptRowProps, ImportAttemptRowState> {
+
+    constructor(props: ImportAttemptRowProps) {
+        super(props);
+        this.state = {
+            open: false,
+            logs: []
+        };
+        this._fetchLogs();
+    }
+
+    /**
+     * Fetches the actual logs using the log IDs, sorts them descendingly by
+     * time_logged, and stores them in this.state.logs.
+     */
+    _fetchLogs() {
+        fetch('/import_attempts/' + this.props.attempt.attempt_id + '/logs').then(response => response.json()).then(data => {
+            data.sort((log1: ProgressLog, log2: ProgressLog) => log1.time_logged > log2.time_logged ? 1 : -1);
+            this.setState({ logs: data });
+        });
+    }
+
+    /**
+     * Generates the drop-down rows, one for each log.
+     */
+    _generateLogRows() {
+        return this.state.logs.map(log => (
+            <TableRow key={log.log_id}>
+                <TableCell align="right">{log.time_logged}</TableCell>
+                <TableCell align="right">{log.level}</TableCell>
+                <TableCell align="left">{log.message}</TableCell>
+            </TableRow>
+        ));
+    }
+
+    render() {
+        return (
+            <React.Fragment>
+                <TableRow>
+                    <TableCell>
+                        <IconButton aria-label="Show Logs" size="small" onClick={() => this.setState({ open: !this.state.open })}>
+                            {this.state.open ? <KeyboardArrowUpIcon /> : <KeyboardArrowDownIcon />}
+                        </IconButton>
+                    </TableCell>
+                    <TableCell align="right">{this.props.attempt.absolute_import_name}</TableCell>
+                    <TableCell align="right">{this.props.attempt.status}</TableCell>
+                    <TableCell align="right">{this.props.attempt.time_created}</TableCell>
+                    <TableCell align="right">{this.props.attempt.time_completed}</TableCell>
+                    <TableCell align="right">{this.props.attempt.provenance_url}</TableCell>
+                    <TableCell align="right">{this.props.attempt.provenance_description}</TableCell>
+                </TableRow>
+                <TableRow>
+                    <TableCell style={{ paddingBottom: 0, paddingTop: 0 }} colSpan={10}>
```

Comment:
consider keeping styling to css files (e.g. padding)

Reply:
created a new file import-table.css. Did I do it correctly?

Reply:
yes! alignment can also be applied the same way (looks like most are right-aligned save for a few). to do that, you can apply a more general rule:
td { text-align: right }

and override it with a more specific rule:
td.my-class { text-align: left }

Reply:
I could not get plain CSS working on the table cells.

```
td {
    text-align: right;
}
```
did not have effects. I switched to the way recommended by the UI framework I am using (material-ui).